src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,83 @@
+import { useContext, FC } from 'react';
+import { DispatchContext, StateContext } from '../../store/todoReducer';
+import { Action } from '../../types/actions';
+import { deleteTodo } from '../../api/todos';
+import { filterAction } from '../../constants/filterActions';
+
+type Props = {
+  onError: (message: string) => void;
+  onCoverShow: React.Dispatch<React.SetStateAction<number[]>>;
+};
+
+export const Footer: FC<Props> = ({ onError, onCoverShow }) => {
+  const { todos, filter } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const isActiveTodo = todos.some(todo => todo.completed);
+
+  const status = todos.filter(todo => !todo.completed);
+
+  const handleCleareCompleted = () => {
+    const completedTodo = todos.filter(todo => todo.completed);
+    const arrOfId = completedTodo.map(todo => todo.id);
+
+    onCoverShow(arrOfId);
+
+    completedTodo.forEach(todo => {
+      deleteTodo(todo.id)
```

Comment:
[Not critical]
The perfect way to handle this would be defining the whole promise chain as a separate function, maybe hook. It might accept list of ids as argument, along with callbacks for catch and finally parts.

Reply:
Something like that:
```
const deleteTodos = ({
  ids,
  onSuccess,
  onError,
  onFnally,
}: {
  ids: string[];
  onSuccess?: (id: string) => void;
  onError?: (id: string) => void;
  onFnally?: (id: string) => void;
}) => {
  ids.forEach(id => {
    deleteTodo(id).then(onSuccess).catch(onError).finally(onFinally);
  });
};
```